### DISCUSSION TITLE
Anyone using a virtual machine to run Gempy? Help needed

### DISCUSSION THREAD

**Discussion by lucianomarquetto** [gempy-project/gempy#809] [Q&A]
Hi all!

I got to a point where my personal computes is struggling to run the model I developed (kernel keeps restarting when adding more surfaces and series or when trying to set resolution higher than 85, 85, 85. Thus, I decided to create a virtual machine in Oracle Cloud, but I could not use it so far. I created a new conda virtual environment but when I tried to pip install Gempy I got an error on pyqt5 installation. 

So my question is, has anyone already tried to use Gempy in a virtual machine, and if so, can you give me some hints on what worked for you? Like what service are you using, how did you set it up, etc. Ideally free stuff, haha!

Talking to a friend he told me this error I'm getting with installing pyqt5 (below) happens because the virtual machine will not work with graphical applications (I could only think of the 3D plotting in Gempy), so I would need a workaround or a different service.

Any help would be appreciated, thanks!

My vm specs are the following:

![image](https://github.com/cgre-aachen/gempy/assets/103533248/4ed187ff-9ad7-461b-a59d-7c101f56e852)

And the error message I get is the following:

> Collecting pyqt5 (from gempy)
> 
>   Using cached PyQt5-5.15.9.tar.gz (3.2 MB)
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... error
>   error: subprocess-exited-with-error
>   
>   × Preparing metadata (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [22 lines of output]
>       Traceback (most recent call last):
>         File "/home/ubuntu/anaconda3/envs/gempy_venv/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
>           main()
>         File "/home/ubuntu/anaconda3/envs/gempy_venv/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
>           json_out['return_val'] = hook(**hook_input['kwargs'])
>         File "/home/ubuntu/anaconda3/envs/gempy_venv/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 152, in prepare_metadata_for_build_wheel
>           whl_basename = backend.build_wheel(metadata_directory, config_settings)
>         File "/tmp/pip-build-env-v1wsye_m/overlay/lib/python3.7/site-packages/sipbuild/api.py", line 47, in build_wheel
>           arguments=_convert_config_settings(config_settings))
>         File "/tmp/pip-build-env-v1wsye_m/overlay/lib/python3.7/site-packages/sipbuild/abstract_project.py", line 87, in bootstrap
>           project.setup(pyproject, tool, tool_description)
>         File "/tmp/pip-build-env-v1wsye_m/overlay/lib/python3.7/site-packages/sipbuild/project.py", line 586, in setup
>           self.apply_user_defaults(tool)
>         File "project.py", line 68, in apply_user_defaults
>           super().apply_user_defaults(tool)
>         File "/tmp/pip-build-env-v1wsye_m/overlay/lib/python3.7/site-packages/pyqtbuild/project.py", line 70, in apply_user_defaults
>           super().apply_user_defaults(tool)
>         File "/tmp/pip-build-env-v1wsye_m/overlay/lib/python3.7/site-packages/sipbuild/project.py", line 237, in apply_user_defaults
>           self.builder.apply_user_defaults(tool)
>         File "/tmp/pip-build-env-v1wsye_m/overlay/lib/python3.7/site-packages/pyqtbuild/builder.py", line 70, in apply_user_defaults
>           "specify a working qmake or add it to PATH")
>       sipbuild.pyproject.PyProjectOptionException
>       [end of output]
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> 
> × Encountered error while generating package metadata.
> ╰─> See above for output.
> 
> note: This is an issue with the package mentioned above, not pip.
> hint: See above for details.

**Comment by AlexanderJuestel** [comment]
@lucianomarquetto I, unfortunately, do not have any experience with that.
